cmd/status/status.go

```diff
@@ -0,0 +1,119 @@
+package status
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"time"
+
+	"numerous.com/cli/internal/app"
+	"numerous.com/cli/internal/appident"
+)
+
+type statusInput struct {
+	appDir  string
+	appSlug string
+	orgSlug string
+}
+
+type appReaderWorkloadLister interface {
+	ReadApp(ctx context.Context, input app.ReadAppInput) (app.ReadAppOutput, error)
+	ListAppWorkloads(ctx context.Context, input app.ListAppWorkloadsInput) ([]app.AppWorkload, error)
+}
+
+func status(ctx context.Context, apps appReaderWorkloadLister, input statusInput) error {
+	ai, err := appident.GetAppIdentifier(input.appDir, nil, input.orgSlug, input.appSlug)
+	if err != nil {
+		appident.PrintGetAppIdentifierError(err, input.appDir, ai)
+		return err
+	}
+
+	readOutput, err := apps.ReadApp(ctx, app.ReadAppInput{OrganizationSlug: ai.OrganizationSlug, AppSlug: ai.AppSlug})
+	if err != nil {
+		app.PrintAppError(err, ai)
+		return err
+	}
+
+	println("Name: " + readOutput.AppDisplayName)
+	if readOutput.AppDescription != "" {
+		println("Description: " + readOutput.AppDescription)
+	}
+
+	workloads, err := apps.ListAppWorkloads(ctx, app.ListAppWorkloadsInput{AppID: readOutput.AppID})
+	if err != nil {
+		app.PrintAppError(err, ai)
+		return err
+	}
+
+	println()
+	if len(workloads) == 0 {
+		println("No workloads found")
+	}
+
+	printWorkloads(workloads)
+
+	return nil
+}
+
+func printWorkloads(workloads []app.AppWorkload) {
+	first := true
+	for _, w := range workloads {
+		if !first {
+			println()
+		}
+		printWorkload(w)
+		first = false
+	}
+}
+
+func printWorkload(workload app.AppWorkload) {
+	if workload.OrganizationSlug != "" {
+		fmt.Printf("Workload in %q:\n", workload.OrganizationSlug)
+	} else if sub := workload.Subscription; sub != nil {
+		fmt.Printf("Workload for subscription %q in %q:\n", sub.SubscriptionUUID, sub.OrganizationSlug)
+	} else {
+		fmt.Println("Workload of unknown origin:")
+	}
+
+	fmt.Printf("    Status: %s\n", workload.Status)
+	fmt.Printf("    Started at: %s (up for %s)\n", workload.StartedAt.Format(time.DateTime), humanDuration(time.Since(workload.StartedAt)))
+}
+
+const (
+	hoursPerDay      float64 = 24.0
+	minutesPerHour   float64 = 60.0
+	secondsPerMinute float64 = 60.0
+)
+
+func humanDuration(since time.Duration) string {
+	hours := since.Hours()
+	if hours > hoursPerDay {
+		fullDays := math.Floor(hours / hoursPerDay)
+		dayHours := math.Floor(hours - fullDays*hoursPerDay)
+		if dayHours > 0.0 {
+			return fmt.Sprintf("%d days and %d hours", int(fullDays), int(dayHours))
+		} else {
+			return fmt.Sprintf("%d days", int(fullDays))
+		}
+	}
+
+	minutes := since.Minutes()
+	if hours > 1.0 {
+		fullHours := math.Floor(hours)
+		hourMinutes := minutes - fullHours*minutesPerHour
```

Comment:
could look something like this, but it's personal preference i guess 😄 

```go
const (
	hoursPerDay      int = 24
	minutesPerHour   int = 60
	secondsPerMinute int = 60
)

func humanDuration(since time.Duration) string {
	hours := int(math.Floor(since.Hours()))
	if hours > hoursPerDay {
		fullDays := hours / hoursPerDay
		remainingHours := hours % hoursPerDay
		if remainingHours > 0 {
			return fmt.Sprintf("%d days and %d hours", fullDays, remainingHours)
		} else {
			return fmt.Sprintf("%d days", fullDays)
		}
	}

	minutes := int(math.Floor(since.Minutes()))
	if hours > 1 {
		fullHours := hours
		remainingMinutes := minutes % minutesPerHour
		if fullHours > 0 {
			return fmt.Sprintf("%d hours and %d minutes", fullHours, remainingMinutes)
		}
	}

	seconds := int(math.Round(since.Seconds()))
	if minutes > 1 {
		fullMinutes := minutes
		remainingSeconds := seconds % secondsPerMinute
		if fullMinutes > 0.0 {
			return fmt.Sprintf("%d minutes and %d seconds", fullMinutes, remainingSeconds)
		}
	}

	return fmt.Sprintf("%d seconds", seconds)
}
```

Reply:
Nice! I updated the PR with your suggestion 😄 I also renamed the function to `humanizeDuration` to be consistent with another function `humanizeBytes`